client/my-sites/site-settings/jetpack-credentials/credentials-form/index.jsx

```diff
@@ -243,6 +243,7 @@ export class CredentialsForm extends Component {
 
 const mapStateToProps = state => ( {
 	formIsSubmitting: isUpdatingJetpackCredentials( state, getSelectedSiteId( state ) ),
+	hostHint: getSelectedSiteSlug( state, getSelectedSiteId( state ) ),
```

Comment:
right here we don't need this `hint` and we don't even need to expose the hint to the React component. if we keep the contract as "we need a host parameter" in the React component we can handle _incidentals_ here in `mapStateToProps`

```js
const mapStateToProps = ( state, { host } ) => Object.assign(
  { formIsSubmitting: … },
  ! host && { host: getSelectedSiteSlug( state, getSelectedSiteId( state ) }
);
```

Reply:
Nice hack! I like it. But, when I do it like this:

```
const mapStateToProps = ( state, { host } ) => Object.assign(
	{ formIsSubmitting: isUpdatingJetpackCredentials( state, getSelectedSiteId( state ) ) },
	! host && { host: getSelectedSiteSlug( state, getSelectedSiteId( state ) ) }
);
```

I get halted on commit with

```
undefined:2
Oops! Something went wrong! :(
^

SyntaxError: Unexpected token O in JSON at position 1
    at Object.parse (native)
    at Socket.process.stdin.on (/Users/rcollier/Code/Calypso/node_modules/eslines/index.js:41:27)
    at emitNone (events.js:91:20)
    at Socket.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
COMMIT ABORTED: The linter reported some problems. If you are aware of them and it is OK, repeat the commit command with --no-verify to avoid this check.
```

Not sure what to make of this. Thoughts?